invictum-test-hybris/src/main/java/com/github/invictum/ImpexImporter.java

```diff
@@ -0,0 +1,99 @@
+            throw new IllegalStateException("Failed to extract URL for import");
+        }
+        String uri = String.format("%s/hac", matcher.group(1));
+        LOG.info("Using {} URI for impex upload", uri);
+        LOG.debug("Using {} key for session", sessionKey);
+        request = new RequestSpecBuilder()
+                .setConfig(RestAssuredConfig.newConfig().sessionConfig(new SessionConfig().sessionIdName(sessionKey)))
+                .setBaseUri(uri).build();
+    }
+
+    private static void makeAuthorizationRequest() {
+        LOG.info("Make authorization request");
+        Response response = RestAssured.given(request).get("/login.jsp");
+        String csrf = response.body().htmlPath().getString("html.head.meta.find { it.@name == '_csrf' }.@content");
+        LOG.debug("CSRF token for authorization is {}", csrf);
+        request.sessionId(response.getSessionId() == null ? "no-session" : response.getSessionId());
+        response = RestAssured.given(request).contentType(ContentType.URLENC).formParameter("_csrf", csrf)
+                .formParameter("j_username", username).formParameter("j_password", password)
+                .post("/j_spring_security_check");
+        request.sessionId(response.getSessionId() == null ? "no-session" : response.getSessionId());
+        LOG.debug("Authorized session is {}", response.getSessionId());
+    }
+
+    private static void waitABit(long milliseconds) {
+        try {
+            LOG.debug("Wait {} milliseconds and retry", milliseconds);
+            Thread.sleep(milliseconds);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void runImport(String impexString) {
+        LOG.info("Make import request");
+        Response response = RestAssured.given(request).get("/console/impex/import");
+        String _csrf = response.body().htmlPath().getString("html.head.meta.find { it.@name == '_csrf' }.@content");
+        response = RestAssured.given(request).param("_csrf", _csrf).param("scriptContent", impexString)
+                .param("validationEnum", "IMPORT_STRICT").param("maxThreads", "1").param("encoding", "UTF-8")
+                .param("_legacyMode", "on").param("enableCodeExecution", "true").param("_enableCodeExecution", "on")
+                .when().post("/console/impex/import");
+        if (attempt == MAX_ATTEMPTS) {
+            attempt = 0;
+            throw new IllegalStateException(String.format("Failed to authorize during %d attempts", MAX_ATTEMPTS));
+        }
+        if (response.statusCode() != 200) {
+            LOG.debug("Import failed. Try to retry {} of {} attempts", attempt, MAX_ATTEMPTS);
+            LOG.debug(response.statusLine());
+            attempt++;
+            if (response.statusCode() == 403) {
+                makeAuthorizationRequest();
+            } else {
+                waitABit(TIMEOUT);
+            }
+            runImport(impexString);
```

Comment:
1) Why you don't use usual cycle approach? At least you will have in logger: 

> Make import request
> Import failed. Try to retry 1 of 5 attempts
> Import failed. Try to retry 2 of 5 attempts
> ...

instead of:

> Make import request
> Import failed. Try to retry 1 of 5 attempts
> Make import request
> Import failed. Try to retry 2 of 5 attempts
> ...

2) Is it possible to configure request, and than, in the cycle just try to post it? For example:
```
request = response = RestAssured.given(request).param("_csrf", _csrf).param("scriptContent", impexString)
.param("validationEnum", "IMPORT_STRICT").param("maxThreads", "1").param("encoding", "UTF-8")
.param("_legacyMode", "on").param("enableCodeExecution", "true").param("_enableCodeExecution", "on")
.when();
do {
     response = request.post();
} while (response.statusCode != 200)
```